### PR TITLE
Cloud Native Chapter 3 Updates

### PR DESCRIPTION
**Pull request by ardalis**
## Summary

Adding missing content.

### CODE REVIEW

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 1]
```diff
@@ -1,13 +1,69 @@
...
+
+Azure API Management (APIM) helps organizations publish APIs in a consistent, manageable fashion. APIM consists of three components: the API Gateway, and administration portal (the Azure portal), and a developer portal.
+
+The API Gateway accepts API calls and routes them to the appropriate backend API. It can also provide additional services like verification of API keys or JWT tokens and API transformation on the fly without code modifications (for instance, to accommodate clients expecting an older interface).
+
+The Azure portal is where you define the API schema and package different APIs into products. You also configure user access, view reports, and configure policies for quotas or transformations.
+
+The developer portal serves as the main resource for developers. It provides developers with API documentation, an interactive test console, and reports on their own usage. Developers also use the portal to create and manage their own accounts, including subscription and API key support.
+
+Using APIM, the eShopOnContainers application exposes several different products of services, each providing a backend for a particular front end client.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]: eShop does not use API Management. For simplicity, we use Ocelot in the implementation but do recommend to use API Management for more complex scenarios. See this: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/architect-microservice-container-applications/direct-client-to-microservice-communication-versus-the-api-gateway-pattern#azure-api-management 
Can you rephrase so it doesn't contradict?

File: docs/architecture/cloud-native/implement-app.md [hunk 2]
```diff
@@ -1,11 +1,11 @@
 ---
-title: Implementing a Cloud Native app
```
Comment (nishanil) [inline docs/architecture/cloud-native/implement-app.md]: Do we need this Section? All that this says is to download the app. I think we can move this to the "Intro" section and drop this .md file. Also, provide a Wiki link for the walkthrough.

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 ---
 title: Introducing eShopOnContainers reference app
-description: Architecting Cloud Native .NET Apps for Azure | Introducing eShopOnContainers Reference App
+description: Architecting Cloud-native .NET Apps for Azure | Introducing eShopOnContainers Reference App
```
Comment (nishanil) [inline docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md]: Per previous review comments from Maira, we need to drop the titles from meta description and describe better for SEO.

File: docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md [hunk 4]
```diff
@@ -52,6 +52,8 @@ Because it leverages microservices, the eShopOnContainers app includes quite a f
 ![Projects in Visual Studio solution.](./media/projects-in-visual-studio-solution.png)
 **Figure 3-4**. Projects in Visual Studio solution.
 
+The code is organized to support the different microservices, and within each microservice, the code is broken up into domain logic, infrastructure concerns, and user interface or service endpoint. In many cases, each service's dependencies can be fulfilled by Azure services in production, as well as alternative options for local development. Let's examine how the application's requirements map to Azure services.
+
```
Comment (nishanil) [inline docs/architecture/cloud-native/introduce-eshoponcontainers-reference-app.md]: When you conclude, mention how this guide focusses on Azure implementation of the sample and for understanding how to architect apps using Microservices, point them to the Microservices book.

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 5]
```diff
@@ -1,13 +1,69 @@
...
 
-(expand on mapping)
+The application's architecture is shown in Figure 3-X. On the left are the client apps, broken up into mobile, traditional Web, and Web Single Page Application (SPA) flavors. On the right are the server-side side components that make up the system, each of which can be hosted in Docker containers and Kubernetes clusters. The traditional web app is powered by the ASP.NET Core MVC application shown in yellow. This app and the mobile and web SPA applications, communicates with the individual microservices through one or more API gateways. The API gateways follow the "backends for front ends" (BFF) pattern, meaning that each gateway is designed to support a given front end client. The individual microservices are listed to the right of the API gateways, and include both business logic and some kind of persistence store. The different services make use of SQL Server databases, Redis cache instances, and MongoDB/CosmosDB stores. On the far right is the system's Event Bus, which is used for communication between the microservices.
+
+![eShopOnContainers Architecture](media/eshoponcontainers-architecture.png)
+**Figure 3-X**. The eShopOnContainers Architecture.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]: Fix figure numbering on image and update sentences which refer them.

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 6]
```diff
@@ -1,13 +1,69 @@
...
+description: Architecting Cloud-native .NET Apps for Azure | Mapping eShopOnContainers to Azure Services
 ms.date: 06/30/2019
 ---
 # Mapping eShopOnContainers to Azure Services
 
-Although not required, Azure is very well-suited to supporting the eShopOnContainers because the project was built to be a cloud native application.
+Although not required, Azure is very well-suited to supporting the eShopOnContainers because the project was built to be a cloud-native application. The application is built with .NET Core, so it's capable of running on Linux or Windows containers depending on the Docker host. The application is made up of multiple autonomous microservices, each with its own data. The different microservices showcase different approaches, ranging from simple CRUD operations to more complex DDD and CQRS patterns. Microservices communicate with clients over HTTP and with one another via message-based communication. The application supports multiple platforms for clients as well, since it leverages HTTP as a standard communication protocol and includes ASP.NET Core apps as well as Xamarin mobile apps that run on Android, iOS, and Windows platforms.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
Although not required, Azure is well-suited to supporting the eShopOnContainers because the project was built to be a cloud-native application. The application is built with .NET Core, so it's capable of running on Linux or Windows containers depending on the Docker host. The application is made up of multiple autonomous microservices, each with its own data. The different microservices showcase different approaches, ranging from simple CRUD operations to more complex DDD and CQRS patterns. Microservices communicate with clients over HTTP and with one another via message-based communication. The application supports multiple platforms for clients as well, since it adopts HTTP as a standard communication protocol and includes ASP.NET Core apps as well as Xamarin mobile apps that run on Android, iOS, and Windows platforms.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 7]
```diff
@@ -1,13 +1,69 @@
...
 
-(expand on mapping)
+The application's architecture is shown in Figure 3-X. On the left are the client apps, broken up into mobile, traditional Web, and Web Single Page Application (SPA) flavors. On the right are the server-side side components that make up the system, each of which can be hosted in Docker containers and Kubernetes clusters. The traditional web app is powered by the ASP.NET Core MVC application shown in yellow. This app and the mobile and web SPA applications, communicates with the individual microservices through one or more API gateways. The API gateways follow the "backends for front ends" (BFF) pattern, meaning that each gateway is designed to support a given front end client. The individual microservices are listed to the right of the API gateways, and include both business logic and some kind of persistence store. The different services make use of SQL Server databases, Redis cache instances, and MongoDB/CosmosDB stores. On the far right is the system's Event Bus, which is used for communication between the microservices.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The application's architecture is shown in Figure 3-X. On the left are the client apps, broken up into mobile, traditional Web, and Web Single Page Application (SPA) flavors. On the right are the server-side components that make up the system, each of which can be hosted in Docker containers and Kubernetes clusters. The traditional web app is powered by the ASP.NET Core MVC application shown in yellow. This app and the mobile and web SPA applications communicate with the individual microservices through one or more API gateways. The API gateways follow the "backends for front ends" (BFF) pattern, meaning that each gateway is designed to support a given front-end client. The individual microservices are listed to the right of the API gateways and include both business logic and some kind of persistence store. The different services make use of SQL Server databases, Redis cache instances, and MongoDB/CosmosDB stores. On the far right is the system's Event Bus, which is used for communication between the microservices.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 8]
```diff
@@ -1,13 +1,69 @@
...
+
+![eShopOnContainers Architecture](media/eshoponcontainers-architecture.png)
+**Figure 3-X**. The eShopOnContainers Architecture.
+
+The server-side components of this architecture all map easily to Azure services.
+
+## Container orchestration and clustering
+
+The application's container-hosted services, from ASP.NET Core MVC apps to individual Catalog and Ordering microservices, can all be hosted and managed in Azure Kubernetes Service (AKS). The application can run locally on Docker and Kubernetes, and the same containers can then be deployed to staging and production environments hosted in AKS. This process can be completely automated as we'll see in the next section.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The application's container-hosted services, from ASP.NET Core MVC apps to individual Catalog and Ordering microservices, can be hosted and managed in Azure Kubernetes Service (AKS). The application can run locally on Docker and Kubernetes, and the same containers can then be deployed to staging and production environments hosted in AKS. This process can be automated as we'll see in the next section.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 9]
```diff
@@ -1,13 +1,69 @@
...
+
+AKS provides management services for individual clusters of containers. The application will deploy separate AKS clusters for each microservice shown in the architecture diagram above. This approach allows each individual service to each independently according to its resource demands. Each microservice can also be deployed independently, and ideally such deployments should incur zero system downtime.
+
+## API Gateway
+
+The eShopOnContainers application has multiple front end clients and multiple different backend services. There is no one-to-one correspondence between the client applications and the microservices that support them. In such a scenario, there may be a great deal of complexity when writing client software to interface with the various backend services in a secure manner. Each client would need to address this complexity on its own, resulting in duplication and many places in which to make updates as services change or new policies are implemented.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The eShopOnContainers application has multiple front-end clients and multiple different backend services. There is no one-to-one correspondence between the client applications and the microservices that support them. In such a scenario, there may be a great deal of complexity when writing client software to interface with the various backend services in a secure manner. Each client would need to address this complexity on its own, resulting in duplication and many places in which to make updates as services change or new policies are implemented.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 10]
```diff
@@ -1,13 +1,69 @@
...
+
+## Data
+
+The various backend services used by eShopOnContainers have different storage requirements. Several microservices use SQL Server databases. The Basket microservice leverages a Redis cache for its persistence. The Locations microservice expects a MongoDB API for its data. Azure supports each of these data formats.
+
+For SQL Server database support, Azure has products for everything from single databases up to highly scalable SQL Database elastic pools. Individual microservices can be configured to communicate with their own individual SQL Server databases quickly and easily. These can be scaled as needed to support each separate microservice according to its needs.
+
+The eShopOnContainers application stores the user's current shopping basket between requests. This is managed by the Basket microservice which stores the data in a Redis cache. In development, this cache can be deployed in a container, while in production in can leverage Azure Cache for Redis. Azure Cache for Redis is a fully managed service offering high performance and reliability without the need to deploy and manage your own Redis instances or containers.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The eShopOnContainers application stores the user's current shopping basket between requests. This is managed by the Basket microservice that stores the data in a Redis cache. In development, this cache can be deployed in a container, while in production it can utilize Azure Cache for Redis. Azure Cache for Redis is a fully managed service offering high performance and reliability without the need to deploy and manage Redis instances or containers on your own.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 11]
```diff
@@ -1,13 +1,69 @@
...
+
+The Locations microservice uses a MongoDB NoSQL database for its persistence. During development, this can be deployed in its own container, while in production the service can leverage [Azure Cosmos DB's API for MongoDB](https://docs.microsoft.com/azure/cosmos-db/mongodb-introduction). One of the benefits of Azure Cosmos DB is its ability to leverage multiple different communication protocols, including a SQL API as well as common NoSQL APIs including MongoDB (as well as Cassandra, Gremlin, and Azure Table Storage). Azure Cosmos DB offers a fully managed and globally distributed database as a service that can scale to meet the needs of the services that use it.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The Locations microservice uses a MongoDB NoSQL database for its persistence. During development, the database can be deployed in its own container, while in production the service can leverage [Azure Cosmos DB's API for MongoDB](https://docs.microsoft.com/azure/cosmos-db/mongodb-introduction). One of the benefits of Azure Cosmos DB is its ability to leverage multiple different communication protocols, including a SQL API as well as common NoSQL APIs including MongoDB (as well as Cassandra, Gremlin, and Azure Table Storage). Azure Cosmos DB offers a fully managed and globally distributed database as a service that can scale to meet the needs of the services that use it.
```

File: docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md [hunk 12]
```diff
@@ -1,13 +1,69 @@
...
+
+Distributed data in cloud-native applications is covered in more detail in [chapter 5](distributed-data.md).
+
+## Event Bus
+
+The application uses events to communicate changes between different services. This functionality can be implemented with a variety of implementations, and locally the eShopOnContainers application uses [RabbitMQ](https://www.rabbitmq.com/). When hosted in Azure, the application would leverage [Azure Service Bus](https://docs.microsoft.com/azure/service-bus/) for its messaging. Azure Service Bus is a fully managed integration message broker that allows applications and services to communicated with one another in a decoupled, reliable, asynchronous manner. Azure Service Bus supports individual queues as well as separate *topics* to support publisher-subscriber scenarios. The eShopOnContainers application would leverage topics with Azure Service Bus to support distributing messages from one microservice to any other microservice that needed to react to a given message.
```
Comment (nishanil) [inline docs/architecture/cloud-native/map-eshoponcontainers-azure-services.md]:
```suggestion
The application uses events to communicate changes between different services. This functionality can be implemented with a variety of implementations, and locally the eShopOnContainers application uses [RabbitMQ](https://www.rabbitmq.com/). When hosted in Azure, the application would leverage [Azure Service Bus](https://docs.microsoft.com/azure/service-bus/) for its messaging. Azure Service Bus is a fully managed integration message broker that allows applications and services to communicate with one another in a decoupled, reliable, asynchronous manner. Azure Service Bus supports individual queues as well as separate *topics* to support publisher-subscriber scenarios. The eShopOnContainers application would leverage topics with Azure Service Bus to support distributing messages from one microservice to any other microservice that needed to react to a given message.
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 13]
```diff
@@ -1,27 +1,42 @@
 ---
 title: Hosting the eShopOnContainers application
-description: Architecting Cloud Native .NET Apps for Azure | Hosting the eShopOnContainers Application
+description: Architecting Cloud-native .NET Apps for Azure | Hosting the eShopOnContainers Application
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: Fix meta description.

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 14]
```diff
@@ -1,27 +1,42 @@
...
+Once the cluster has been created and configured, you can deploy the application to it using Helm and Tiller.
 
 ## Helm Deployment
 
-asdf
+Basic deployments to AKS may use custom CLI scripts or simple deployment files, but more complex applications should use a dependency management tool like Helm. Helm is maintained by the Cloud-native Computing Foundation and helps you define, install, and upgrade Kubernetes applications. Helm is composed of a command line client, helm, which uses helm charts, and an in-cluster component, Tiller. Helm Charts use standard YAML-formatted files to describe a related set of Kubernetes resources, and are typically versioned alongside the application they describe. Helm Charts range from extremely simple to very complex depending on the requirements of the installation they describe.
+
+You'll find the eShopOnContainers helm charts in the /k8s/helm folder. Figure 3-X shows how the different components of the application are organized into a folder structure used by helm to define and managed deployments.
+
+![eShopOnContainers Architecture](media/eshoponcontainers-helm-folder.png)
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: Remove the mouse pointer from the screenshot.

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 15]
```diff
@@ -1,27 +1,42 @@
...
+**Figure 3-X**. The eShopOnContainers helm folder.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: Fix Figure Number

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 16]
```diff
@@ -1,27 +1,42 @@
...
 
 ## Azure Kubernetes Service
 
-asdf
+If you'd like to host the eShopOnContainers application in your own AKS cluster, the first step is to create your cluster. You can do this using the Azure Portal, which will walk you through the required steps, or you can use the Azure CLI, taking care to ensure you enable Role-Based Access Control (RBAC) and application routing when you do so. The eShopOnContainers' documentation describes the steps involved in creating your own AKS cluster. Once the cluster is created, you must enable access to the Kubernetes dashboard, at which point you should be able to browse to the Kubernetes dashboard to manage the cluster.
+
+Once the cluster has been created and configured, you can deploy the application to it using Helm and Tiller.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: You can link to this WIKI: https://github.com/dotnet-architecture/eShopOnContainers/wiki/Deploy-to-Azure-Kubernetes-Service-(AKS) for deploying eShop on AKS

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 17]
```diff
@@ -1,27 +1,42 @@
...
 
 ## Helm Deployment
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
## Deploying to Azure Kubernetes Service using Helm
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 18]
```diff
@@ -1,27 +1,42 @@
 ---
 title: Hosting the eShopOnContainers application
-description: Architecting Cloud Native .NET Apps for Azure | Hosting the eShopOnContainers Application
+description: Architecting Cloud-native .NET Apps for Azure | Hosting the eShopOnContainers Application
 ms.date: 06/30/2019
 ---
 # Hosting the eShopOnContainers application
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
# Deploying eShopOnContainers to Azure
```
How does this title sound?

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 19]
```diff
@@ -1,27 +1,42 @@
...
+
+Once the cluster has been created and configured, you can deploy the application to it using Helm and Tiller.
 
 ## Helm Deployment
 
-asdf
+Basic deployments to AKS may use custom CLI scripts or simple deployment files, but more complex applications should use a dependency management tool like Helm. Helm is maintained by the Cloud-native Computing Foundation and helps you define, install, and upgrade Kubernetes applications. Helm is composed of a command line client, helm, which uses helm charts, and an in-cluster component, Tiller. Helm Charts use standard YAML-formatted files to describe a related set of Kubernetes resources, and are typically versioned alongside the application they describe. Helm Charts range from extremely simple to very complex depending on the requirements of the installation they describe.
+
+You'll find the eShopOnContainers helm charts in the /k8s/helm folder. Figure 3-X shows how the different components of the application are organized into a folder structure used by helm to define and managed deployments.
+
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: There has been a lot of interests for an explanation on Helm in our WIKIs. I think we should elaborate more on Helm, Tiller, Helm Charts in separate paragraphs. Also, it is not clear how Helm helps to manage complex deployments.
You can also mention how Helm simplifies deployment steps in release pipelines of DevOps.

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 20]
```diff
@@ -1,27 +1,42 @@
...
+![eShopOnContainers Architecture](media/eshoponcontainers-helm-folder.png)
+**Figure 3-X**. The eShopOnContainers helm folder.
+
+Each individual component is installed using a `helm install` command. These commands is easily scripted, and eShopOnContainers provides a "deploy all" script that loops through the different components and installs them using their respective helm charts. The result is a repeatable process, versioned with the application in source control, that anyone on the team can deploy to an AKS cluster with a one line script command. Especially when combined with DevSpaces, this makes it very easy for developers to diagnose and test their individual changes to their microservice-based cloud-native apps.
 
 ## DevSpaces for Kubernetes
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
## Azure Dev Spaces
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 21]
```diff
@@ -1,27 +1,42 @@
...
 
-asdf
+Azure Dev Spaces helps individual developers host their own unique version of AKS clusters in Azure during development. This minimize local machine requirements and allows team members to quickly see how their changes will behave in a real AKS environment. Azure Dev Spaces offers a CLI for developers to use to manage their dev spaces and to deploy to a specific child dev space as needed. Each child dev space is referenced using a unique URL subdomain, allowing side-by-side deployments of modified clusters so that individual developers can avoid conflicting with each other's work in progress.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]: Can you add the Image that explains how developers can use Azure Dev Spaces? We have to improve this explanation here since we don't have another section to explain this in detail.

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 22]
```diff
@@ -1,27 +1,42 @@
...
 ms.date: 06/30/2019
 ---
 # Hosting the eShopOnContainers application
 
-The logic supporting the eShopOnContainers application can be supported by Azure using a variety of services. The recommended approach is to leverage Kubernetes using Azure Kubernetes Service. This can be combined with Helm deployment to ensure easily repeated infrastructure configuration. Optionally, developers can leverage DevSpaces for Kubernetes as part of their development process. Another option is to host the functionality of the app using Azure Serverless features like Azure Functions and Azure Logic Apps.
+The logic supporting the eShopOnContainers application can be supported by Azure using a variety of services. The recommended approach is to leverage Kubernetes using Azure Kubernetes Service (AKS). This can be combined with Helm deployment to ensure easily repeated infrastructure configuration. Optionally, developers can leverage DevSpaces for Kubernetes as part of their development process. Another option is to host the functionality of the app using Azure Serverless features like Azure Functions and Azure Logic Apps.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
The logic supporting the eShopOnContainers application can be supported by Azure using a variety of services. The recommended approach is to leverage Kubernetes using Azure Kubernetes Service (AKS). This can be combined with Helm deployment to ensure easily repeated infrastructure configuration. Optionally, developers can leverage Azure Dev Spaces for Kubernetes as part of their development process. Another option is to host the functionality of the app using Azure Serverless features like Azure Functions and Azure Logic Apps.
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 23]
```diff
@@ -1,27 +1,42 @@
...
 
 ## Azure Kubernetes Service
 
-asdf
+If you'd like to host the eShopOnContainers application in your own AKS cluster, the first step is to create your cluster. You can do this using the Azure Portal, which will walk you through the required steps, or you can use the Azure CLI, taking care to ensure you enable Role-Based Access Control (RBAC) and application routing when you do so. The eShopOnContainers' documentation describes the steps involved in creating your own AKS cluster. Once the cluster is created, you must enable access to the Kubernetes dashboard, at which point you should be able to browse to the Kubernetes dashboard to manage the cluster.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
If you'd like to host the eShopOnContainers application in your own AKS cluster, the first step is to create your cluster. You can do this using the Azure portal, which will walk you through the required steps, or you can use the Azure CLI, taking care to ensure you enable Role-Based Access Control (RBAC) and application routing when you do so. The eShopOnContainers' documentation describes the steps involved in creating your own AKS cluster. Once the cluster is created, you must enable access to the Kubernetes dashboard, at which point you should be able to browse to the Kubernetes dashboard to manage the cluster.
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 24]
```diff
@@ -1,27 +1,42 @@
...
+
+Once the cluster has been created and configured, you can deploy the application to it using Helm and Tiller.
 
 ## Helm Deployment
 
-asdf
+Basic deployments to AKS may use custom CLI scripts or simple deployment files, but more complex applications should use a dependency management tool like Helm. Helm is maintained by the Cloud-native Computing Foundation and helps you define, install, and upgrade Kubernetes applications. Helm is composed of a command line client, helm, which uses helm charts, and an in-cluster component, Tiller. Helm Charts use standard YAML-formatted files to describe a related set of Kubernetes resources, and are typically versioned alongside the application they describe. Helm Charts range from extremely simple to very complex depending on the requirements of the installation they describe.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
Basic deployments to AKS may use custom CLI scripts or simple deployment files, but more complex applications should use a dependency management tool like Helm. Helm is maintained by the Cloud-native Computing Foundation and helps you define, install, and upgrade Kubernetes applications. Helm is composed of a command-line client, helm, which uses helm charts, and an in-cluster component, Tiller. Helm Charts use standard YAML-formatted files to describe a related set of Kubernetes resources and are typically versioned alongside the application they describe. Helm Charts range from simple to complex depending on the requirements of the installation they describe.
```

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md [hunk 25]
```diff
@@ -1,27 +1,42 @@
...
+
+You'll find the eShopOnContainers helm charts in the /k8s/helm folder. Figure 3-X shows how the different components of the application are organized into a folder structure used by helm to define and managed deployments.
+
+![eShopOnContainers Architecture](media/eshoponcontainers-helm-folder.png)
+**Figure 3-X**. The eShopOnContainers helm folder.
+
+Each individual component is installed using a `helm install` command. These commands is easily scripted, and eShopOnContainers provides a "deploy all" script that loops through the different components and installs them using their respective helm charts. The result is a repeatable process, versioned with the application in source control, that anyone on the team can deploy to an AKS cluster with a one line script command. Especially when combined with DevSpaces, this makes it very easy for developers to diagnose and test their individual changes to their microservice-based cloud-native apps.
```
Comment (nishanil) [inline docs/architecture/cloud-native/host-eshoponcontainers-application.md]:
```suggestion
Each individual component is installed using a `helm install` command. These commands are easily scripted, and eShopOnContainers provides a "deploy all" script that loops through the different components and installs them using their respective helm charts. The result is a repeatable process, versioned with the application in source control, that anyone on the team can deploy to an AKS cluster with a one line script command. Especially when combined with DevSpaces, this makes it easy for developers to diagnose and test their individual changes to their microservice-based cloud-native apps.
```

**Review by nishanil — COMMENTED** [review]
@ardalis Let a few comments for you to address. Some of the suggestions are fixes to improve Acrolinx score.

File: docs/architecture/cloud-native/host-eshoponcontainers-application.md (hunk 21 above)
Comment (nishanil) [reply docs/architecture/cloud-native/host-eshoponcontainers-application.md]: The one in the intro chapter was removed. It was too soon to talk about that in length. What section are you planning to have this covered?

**Comment by ardalis** [comment]
@nishanil I need to expand on Helm but I think that's the only thing left. Can we merge this PR, let me swap chapters 2 and 3, and I'll add more about Helm and its advantages after? I need to do some research there and I want to get the swap done and some other cleanup tasks without waiting on that.

**Comment by nishanil** [comment]
@ardalis Can you fix the TOC as well?
```
Line 0: [Warning] Invalid file link:(~/docs/architecture/cloud-native/implement-app.md).
Line 0: [Warning] Invalid file link:(~/docs/architecture/cloud-native/host-eshoponcontainers-application.md).
```

**Comment by ardalis** [comment]
done. I always forget that one... :)

**Review by nishanil — APPROVED** [review]
